### PR TITLE
update assignment op to work with partially bound variables

### PR DESCRIPTION
**Pull request by leina05**
PR checklist:

- [ ] Added changelog entry.

### CODE REVIEW

File: polar-core/src/vm.rs [hunk 1]
```diff
@@ -1452,20 +1452,7 @@ impl PolarVirtualMachine {
...
-                            return Err(self.type_error(
-                                &left,
-                                format!(
-                                    "Can only assign to unbound variables, {} is not unbound.",
-                                    var.to_polar()
-                                ),
-                            ));
-                        }
-                    },
+                    (Value::Variable(var), _) => self.bind(var, right)?,
```
Comment (dhatch) [inline polar-core/src/vm.rs:1455]: If the intention is to fail on bound values, we need to first check the variable state. If `right` is an unbound variable and `var` is bound `bind` will bind `right` to the value of `var`.

With two variable arguments `bind` binds them together and ensures they have the same value, it is bi-directional.

(This part of the interface was a bit up for debate, so if it's counterintuitive we could change that)

File: polar-core/tests/integration_tests.rs [hunk 2]
```diff
@@ -1711,8 +1711,13 @@ fn test_assignment() {
...
-        s == "Can only assign to unbound variables, x is not unbound."
+        RuntimeError::IncompatibleBindings { msg: s },
+        s == "Cannot rebind Symbol(\"x\")"
+    );
+    qruntime!(
+        "x < 3 and x := 6",
+        RuntimeError::IncompatibleBindings { msg: s },
+        s == "Grounding failed"
     );
     qnull(&mut p, "x := 5 and x > 6");
```
Comment (dhatch) [inline polar-core/tests/integration_tests.rs:1722]: Add a test case like `x = 1 and x := y`. This should fail still

**Review by dhatch — CHANGES_REQUESTED** [review]

File: polar-core/src/vm.rs (hunk 1 above)
Comment (leina05) [reply polar-core/src/vm.rs:1455]: oh, interesting. Okay

**Comment by dhatch** [comment]
I am going to add a task to the next cycle to continue the conversation on how this should behave & get something merged to fix it if needed.

**Pull request closed without merging**